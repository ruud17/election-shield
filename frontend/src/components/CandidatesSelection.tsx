import { FC, ChangeEvent } from "react";
import { Row, Alert, Form, Card, Col } from "react-bootstrap";
import { Candidates } from "../utils/enum";
import DonaldImg from "../images/Donald.jpeg";
import JoeImg from "../images/Joe.jpeg";

export interface CandidatesSelectionProps {
  selectedCandidate: string;
  handleOptionChange: (event: ChangeEvent<HTMLInputElement>) => void;
}

const CandidatesSelection: FC<CandidatesSelectionProps> = ({
  selectedCandidate,
  handleOptionChange,
}) => {
  return (
    <Row className="candidate">
      <Alert variant="info">
        Please select your preferred presidential candidate
      </Alert>
      <Col>
        <Form.Group>
          <Card className="bg-light text-black">
            <Card.Img variant="top" src={DonaldImg} />
            <Card.Body>
              <Card.Title>
                <Form.Check
                  type="radio"
                  label={Candidates.DonaldTrump}
                  value={Candidates.DonaldTrump}
                  checked={selectedCandidate === Candidates.DonaldTrump}
                  onChange={handleOptionChange}
                />
              </Card.Title>
            </Card.Body>
          </Card>
        </Form.Group>
      </Col>
      <Col>
        <Form.Group>
          <Card className="bg-light text-black">
            <Card.Img variant="top" src={JoeImg} />
            <Card.Body>
              <Card.Title>
                <Form.Check
                  type="radio"
                  label={Candidates.JoeBiden}
                  value={Candidates.JoeBiden}
                  checked={selectedCandidate === Candidates.JoeBiden}
                  onChange={handleOptionChange}
                />
              </Card.Title>
            </Card.Body>
          </Card>
        </Form.Group>
      </Col>
    </Row>
  );
};

export default CandidatesSelection;
