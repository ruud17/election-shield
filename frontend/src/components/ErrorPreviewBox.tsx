import { FC } from "react";
import { Row, Alert } from "react-bootstrap";
import { ViewError } from "../utils/error";

interface ErrorPreviewBoxProps {
  errors: unknown[];
}

const ErrorPreviewBox: FC<ErrorPreviewBoxProps> = ({ errors }) => {
  return (
    <Row>
      {errors.map((err, ind) => (
        <Alert variant="danger" key={ind}>
          <p className="mb-0">{ViewError.makeNewErrorToDisplay(err)}</p>
        </Alert>
      ))}
    </Row>
  );
};

export default ErrorPreviewBox;
