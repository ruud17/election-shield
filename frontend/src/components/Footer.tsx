import { FC } from "react";
import { Container, Navbar } from "react-bootstrap";

const Footer: FC = () => {
  return (
    <Navbar fixed="bottom" bg="dark" variant="dark" className="footer">
      <Navbar.Text>&copy; {new Date().getFullYear()} Mehmed Jusic</Navbar.Text>
    </Navbar>
  );
};

export default Footer;
