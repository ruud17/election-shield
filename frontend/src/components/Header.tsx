import { FC } from "react";
import { Navbar, Image } from "react-bootstrap";
import VoteLogo from "../images/vote.svg";

const Header: FC = () => {
  return (
    <Navbar fixed="top" bg="dark" data-bs-theme="light" className="header">
      <Image src={VoteLogo} className="election-logo " />
      Election Shield
    </Navbar>
  );
};

export default Header;
