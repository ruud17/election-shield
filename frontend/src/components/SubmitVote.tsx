import { FC } from "react";
import { Container, Row, Col, Button, Form } from "react-bootstrap";

const SubmitVote: FC = () => {
  return (
    <Row className="submit-vote">
      <Col>
        <Form.Check type="checkbox">
          <Form.Check.Input type="checkbox" />
          <Form.Check.Label>
            Accept Terms of Service and Data Collection Consent
          </Form.Check.Label>
        </Form.Check>
      </Col>
      <Col className="d-flex justify-content-end">
        <Button variant="primary" size="sm">
          Submit
        </Button>
      </Col>
    </Row>
  );
};

export default SubmitVote;
