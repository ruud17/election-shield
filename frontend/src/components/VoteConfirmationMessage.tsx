import { FC } from "react";
import { Container } from "react-bootstrap";

interface VoteConfirmationMessageProps {
  // name: string;
}

const VoteConfirmationMessage: FC<VoteConfirmationMessageProps> = () => {
  return <Container>Thank you for successfull voting!!!</Container>;
};

export default VoteConfirmationMessage;
