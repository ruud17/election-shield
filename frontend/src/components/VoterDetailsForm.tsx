import { FC, ChangeEvent } from "react";
import { Row, Col, Form, FloatingLabel } from "react-bootstrap";
import { VoterDetails } from "../utils/interfaces";

export interface VoterDetailsFormProps {
  voterDetails: VoterDetails;
  handleInputChange: (event: ChangeEvent<HTMLInputElement>) => void;
  formErrors: any;
}

const VoterDetailsForm: FC<VoterDetailsFormProps> = ({
  voterDetails,
  handleInputChange,
  formErrors,
}) => {
  return (
    <Form className="voter-details-form">
      <Row>
        <Col>
          <FloatingLabel
            controlId="firstName"
            label="First Name"
            className="mb-3"
          >
            <Form.Control
              type="text"
              placeholder="First Name"
              size="sm"
              name="firstName"
              value={voterDetails.firstName}
              onChange={handleInputChange}
            />
            <Form.Control.Feedback type="invalid">
              {formErrors.firstName}
            </Form.Control.Feedback>
          </FloatingLabel>
        </Col>

        <Col>
          <FloatingLabel
            controlId="lastName"
            label="Last Name"
            className="mb-3"
          >
            <Form.Control
              type="text"
              placeholder="Last Name"
              size="sm"
              name="lastName"
              value={voterDetails.lastName}
              onChange={handleInputChange}
            />
            <Form.Control.Feedback type="invalid">
              {formErrors.lastName}
            </Form.Control.Feedback>
          </FloatingLabel>
        </Col>
      </Row>
      <Row>
        <Col>
          <FloatingLabel controlId="email" label="Email" className="mb-3">
            <Form.Control
              type="email"
              placeholder="Email"
              size="sm"
              name="email"
              value={voterDetails.email}
              onChange={handleInputChange}
            />
            <Form.Control.Feedback type="invalid">
              {formErrors.email}
            </Form.Control.Feedback>
          </FloatingLabel>
        </Col>
        <Col>
          <FloatingLabel
            controlId="identity_id"
            label="Identity Id"
            className="mb-3"
          >
            <Form.Control
              type="string"
              placeholder="Identity Id"
              aria-describedby="idHelpBlock"
              size="sm"
              name="identity_id"
              value={voterDetails.identity_id}
              onChange={handleInputChange}
            />
            <Form.Control.Feedback type="invalid">
              {formErrors.identity_id}
            </Form.Control.Feedback>
          </FloatingLabel>
        </Col>
      </Row>
    </Form>
  );
};

export default VoterDetailsForm;
