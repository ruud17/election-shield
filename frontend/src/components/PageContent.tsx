import { FC, useState, ChangeEvent } from "react";
import { Container } from "react-bootstrap";
import VoterDetailsForm from "./VoterDetailsForm";
import CandidatesSelection from "./CandidatesSelection";
import SubmitVote from "./SubmitVote";
import { VoterDetails } from "../utils/interfaces";
import { initialVoterDetailsData } from "../utils/defaultStates";
import { Candidates } from "../utils/enum";

const PageContent: FC = () => {
  const [voterDetails, setVoterDetails] = useState<VoterDetails>(
    initialVoterDetailsData
  );

  const [selectedCandidate, setSelectedCandidate] = useState<string>(
    Candidates.DonaldTrump
  );

  const [isTermsCheckboxConfirmed, setIsTermsCheckboxConfirmed] =
    useState<boolean>(false);

  const handleVoterDetailChange = (event: ChangeEvent<HTMLInputElement>) => {
    const { name, value } = event.target;
    setVoterDetails({
      ...voterDetails,
      [name]: value,
    });
  };

  const handleCandidateSelectionChange = (
    event: ChangeEvent<HTMLInputElement>
  ) => {
    setSelectedCandidate(event.target.value);
  };

  const handleTermsOfServiceOptionConfirmation = () => {
    setIsTermsCheckboxConfirmed(!isTermsCheckboxConfirmed);
  };

  return (
    <Container className="page-content">
      <VoterDetailsForm
        voterDetails={voterDetails}
        handleInputChange={handleVoterDetailChange}
      />
      <CandidatesSelection
        selectedCandidate={selectedCandidate}
        handleOptionChange={handleCandidateSelectionChange}
      />
      <SubmitVote
        isTermsCheckboxConfirmed={isTermsCheckboxConfirmed}
        handleChange={handleTermsOfServiceOptionConfirmation}
      />
    </Container>
  );
};

export default PageContent;
