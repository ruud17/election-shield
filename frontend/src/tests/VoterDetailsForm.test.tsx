import renderer from "react-test-renderer";
import VoterDetailsForm from "../components/VoterDetailsForm";
import { VoterDetailsFormProps } from "../components/VoterDetailsForm";

const componentProps: VoterDetailsFormProps = {
  voterDetails: {
    email: "johndoe@test.com",
    firstName: "John",
    lastName: "Doe",
    identity_id: "1234567",
  },
  handleInputChange: () => {},
  formErrors: { lastName: "Required" },
};

test("renders correctly", () => {
  const tree = renderer
    .create(<VoterDetailsForm {...componentProps} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
