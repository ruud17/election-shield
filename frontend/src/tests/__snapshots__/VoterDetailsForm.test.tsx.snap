// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="voter-details-form"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="mb-3 form-floating"
      >
        <input
          className="form-control form-control-sm"
          id="firstName"
          name="firstName"
          onChange={[Function]}
          placeholder="First Name"
          type="text"
          value="John"
        />
        <div
          className="invalid-feedback"
        />
        <label
          htmlFor="firstName"
        >
          First Name
        </label>
      </div>
    </div>
    <div
      className="col"
    >
      <div
        className="mb-3 form-floating"
      >
        <input
          className="form-control form-control-sm"
          id="lastName"
          name="lastName"
          onChange={[Function]}
          placeholder="Last Name"
          type="text"
          value="Doe"
        />
        <div
          className="invalid-feedback"
        >
          Required
        </div>
        <label
          htmlFor="lastName"
        >
          Last Name
        </label>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="mb-3 form-floating"
      >
        <input
          className="form-control form-control-sm"
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="Email"
          type="email"
          value="johndoe@test.com"
        />
        <div
          className="invalid-feedback"
        />
        <label
          htmlFor="email"
        >
          Email
        </label>
      </div>
    </div>
    <div
      className="col"
    >
      <div
        className="mb-3 form-floating"
      >
        <input
          aria-describedby="idHelpBlock"
          className="form-control form-control-sm"
          id="identity_id"
          name="identity_id"
          onChange={[Function]}
          placeholder="Identity Id"
          type="string"
          value="1234567"
        />
        <div
          className="invalid-feedback"
        />
        <label
          htmlFor="identity_id"
        >
          Identity Id
        </label>
      </div>
    </div>
  </div>
</form>
`;
